src/jit/gentree.cpp

```diff
@@ -14112,7 +14112,7 @@ GenTree* Compiler::gtFoldExprConst(GenTree* tree)
                     {
                         f1 = forceCastToFloat(d1);
                         f2 = forceCastToFloat(d2);
-                        d1 = f1 - f2;
+                        d1 = forceCastToFloat(f1 - f2);
```

Comment:
Would it be beneficial to just store these as floats as well?

Reply:
(maybe as a long term goal, not necessarily as part of this PR)

Reply:
Hard to say ... there are some interesting bits of code here and there which might benefit:

https://github.com/dotnet/coreclr/blob/8939f019c34febbf0bab9dcc0f13b9ddc58735d8/src/jit/optcse.cpp#L2506-L2520

Reply:
Right, and in apps that largely use `floats`, rather than `doubles` (like ML.NET, games, graphics, etc); there might be some space-savings (and with the `System.MathF` support, you don't have the intermediate `double` either).

Reply:
> There's no connection between storing float constants as doubles, JIT32 leftovers like the CSE one above and space-savings of any kind.

I meant that `double` takes twice the space of `float` (8-bytes vs 4-bytes), so each `float` constant being stored is taking twice as much memory as is needed. That will have some impact in the tree, everywhere a `DconNode` appears, and in the ValueNum store where the values are actually cached (although this looks to already have separate maps for double vs float).

Reply:
> I meant that double takes twice the space of float (8-bytes vs 4-bytes), so each float constant being stored is taking twice as much memory as is needed. That will have some impact in the tree, everywhere a DconNode appears,

No, it won't.

Reply:
> No, it won't.

Would you care to elaborate? There is obviously something I'm missing here 😄

Reply:
> Would you care to elaborate? There is obviously something I'm missing here 

Like the fact that tree nodes have a fixed size so removing stuff from them won't make them smaller. And that on 64 bit targets using float instead of double will only buy you 4 bytes of padding at the end of the struct.

Reply:
Ah, right. I forgot that it uses `TREE_NODE_SZ_SMALL` and `TREE_NODE_SZ_LARGE`, rather than the sizeof the struct.